google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpcTest.java

```diff
@@ -502,6 +505,33 @@ public void testAdminRequestsLimitExceededRetryAlgorithm() {
     assertThat(alg.shouldRetry(new Exception("random exception"), null)).isFalse();
   }
 
+  @Test
+  public void testCustomUserAgent() {
+    for (final String headerId : new String[] {"user-agent", "User-Agent", "USER-AGENT"}) {
+      final HeaderProvider userAgentHeaderProvider =
+          new HeaderProvider() {
+            @Override
+            public Map<String, String> getHeaders() {
+              final Map<String, String> headers = new HashMap<>();
+              headers.put(headerId, "test-agent");
+              return headers;
+            }
+          };
+      final SpannerOptions options =
+          createSpannerOptions().toBuilder().setHeaderProvider(userAgentHeaderProvider).build();
+      final Spanner spanner = options.getService();
```

Comment:
This `Spanner` instance is never closed, which is (probably) what is causing the build error.

Reply:
Thanks, fixed